Neos.Neos/Classes/Fusion/Helper/NodeHelper.php

```diff
@@ -53,10 +54,13 @@ public function nearestContentCollection(Node $node, string $nodePath): Node
                     $contentCollectionType
                 ), 1409300545);
             }
-            $subNode = $this->findNodeByNodePath(
-                $node,
-                NodePath::fromString($nodePath)
-            );
+            $nodePath = AbsoluteNodePath::tryFromString($nodePath)
+                ?: NodePath::fromString($nodePath);
```

Comment:
This is quite implicit and uses exceptions for control flow. What about something like:
```suggestion
            $nodePath = AbsoluteNodePath::isAbsolutePath($nodePath) ? AbsoluteNodePath::fromString($nodePath) : NodePath::fromString($nodePath);
```
(or maybe a better method name than `isAbsolutePath()`).

Apart from that, static analysis should complain about changing the type of `$nodePath` from `string` to `AbsoluteNodePath|NodePath` – it should get a new variable name.

And while we're on it, we could remove the whole `else` block since there is an early return in place..

In general, this method is quite overcomplex and could be simplified to
```php
public function nearestContentCollection(Node $node, string $nodePath): Node
{
    $contentCollectionType = 'Neos.Neos:ContentCollection';
    if ($node->nodeType->isOfType($contentCollectionType)) {
        return $node;
    }
    if ($nodePath === '') {
        throw new Exception(sprintf(
            'No content collection of type %s could be found in the current node and no node path was provided.'
            . ' You might want to configure the nodePath property'
            . ' with a relative path to the content collection.',
            $contentCollectionType
        ), 1409300545);
    }
    $subgraph = $this->contentRepositoryRegistry->subgraphForNode($node);
    if (AbsoluteNodePath::isAbsolutePath($nodePath)) {
        $subNode = $subgraph->findNodeByAbsolutePath(AbsoluteNodePath::fromString($nodePath));
    } else {
        $subNode = $subgraph->findNodeByPath(NodePath::fromString($nodePath));
    }
    if ($subNode === null) {
        throw new Exception(...);
    }
    if (!$subNode->nodeType->isOfType($contentCollectionType)) {
        throw new Exception(...);
    }
    return $subNode;
}
```

But that makes me wonder about the API.. If you specify an absolute `$nodePath` the `$node` argument is completely ignored (unless it is a content collection itself).
This screams "instability" :)